hedera-mirror-rosetta/app/services/construction/construction_service_test.go

```diff
@@ -0,0 +1,457 @@
+}
+
+func TestNewConstructionAPIService(t *testing.T) {
+	constructionService := NewConstructionAPIService()
+	assert.IsType(t, &ConstructionAPIService{}, constructionService)
+}
+
+func TestConstructionCombine(t *testing.T) {
+	// given:
+	expectedConstructionCombineResponse := &types.ConstructionCombineResponse{
+		SignedTransaction: validTxHash,
+	}
+
+	// when:
+	res, e := NewConstructionAPIService().ConstructionCombine(nil, dummyConstructionCombineRequest())
+
+	// then:
+	assert.Equal(t, expectedConstructionCombineResponse, res)
+	assert.Nil(t, e)
+}
+
+func TestConstructionCombineThrowsWhenNoSingleSignature(t *testing.T) {
+	// given:
+	exampleConstructionCombineRequest := dummyConstructionCombineRequest()
+	exampleConstructionCombineRequest.Signatures = []*types.Signature{}
+
+	// when:
+	res, e := NewConstructionAPIService().ConstructionCombine(nil, exampleConstructionCombineRequest)
+
+	// then:
+	assert.Nil(t, res)
+	assert.Equal(t, errors.Errors[errors.MultipleSignaturesPresent], e)
+}
+
+func TestConstructionCombineThrowsWhenDecodeStringFails(t *testing.T) {
+	// given:
+	exampleInvalidTxHashConstructionCombineRequest := dummyConstructionCombineRequest()
+	exampleInvalidTxHashConstructionCombineRequest.UnsignedTransaction = invalidTxHash
+
+	// when:
+	res, e := NewConstructionAPIService().ConstructionCombine(nil, exampleInvalidTxHashConstructionCombineRequest)
+
+	// then:
+	assert.Nil(t, res)
+	assert.Equal(t, errors.Errors[errors.TransactionDecodeFailed], e)
+}
+
+func TestConstructionCombineThrowsWithInvalidPublicKey(t *testing.T) {
+	// given:
+	exampleInvalidPublicKeyConstructionCombineRequest := dummyConstructionCombineRequest()
+	exampleInvalidPublicKeyConstructionCombineRequest.Signatures[0].PublicKey = &types.PublicKey{}
+
+	// when:
+	res, e := NewConstructionAPIService().ConstructionCombine(nil, exampleInvalidPublicKeyConstructionCombineRequest)
+
+	// then:
+	assert.Nil(t, res)
+	assert.Equal(t, errors.Errors[errors.InvalidPublicKey], e)
+}
+
+func TestConstructionCombineThrowsWhenSignatureIsNotVerified(t *testing.T) {
+	// given:
+	exampleInvalidSigningPayloadConstructionCombineRequest := dummyConstructionCombineRequest()
+	exampleInvalidSigningPayloadConstructionCombineRequest.Signatures[0].SigningPayload = &types.SigningPayload{}
+
+	// when:
+	res, e := NewConstructionAPIService().ConstructionCombine(nil, exampleInvalidSigningPayloadConstructionCombineRequest)
+
+	// then:
+	assert.Nil(t, res)
+	assert.Equal(t, errors.Errors[errors.InvalidSignatureVerification], e)
+}
+
+func TestConstructionDerive(t *testing.T) {
+	// when:
+	res, e := NewConstructionAPIService().ConstructionDerive(nil, nil)
+
+	// then:
+	assert.Nil(t, res)
+	assert.Equal(t, errors.Errors[errors.NotImplemented], e)
+}
+
+func TestConstructionHash(t *testing.T) {
+	// given:
+	exampleConstructionHashRequest := dummyConstructionHashRequest(validTxHash)
+	exampleConstructionHashResponse := &types.TransactionIdentifierResponse{
```

Comment:
This is not actually an example hash response but the `expected` hash response. I think that a better name for the variable would be:
`expectedConstructHashResponse`